### ISSUE TITLE
Window Shade capability

### ISSUE THREAD

**Issue by ericterstegen** [DanielOgorchock/ST_Anything#124]
I am looking to automate some roller blinds but it is very low level.  Basically, ESP32 with a motor driver and a open and closed switch (probably hall sensors).  I noticed ST has a Window Shade capability which would probably make sense for this application.  I am new to ST_Anything and ST in general but I think I could get it working without too much fuss.  Is there plans to have this capability or what do you think?

**Comment by ogiewon** [comment]
I haven’t considered adding the Window Shade capability, as my home has plantation shutters. Please feel free to add a new Arduino c++ class and child device handler via a pull request. Let me know if you have any questions.

**Comment by ericterstegen** [comment]
I will need to be able to run the shades with physical buttons if smartthings is down or as a secondary way to move the shades.  Is there any way to handle local events/interrupts without using the smartthings triggers?

**Comment by ericterstegen** [comment]
I am not sure that is going to work.  It looks like it might be pushing a button with a servo?  This is very low level... I am making everything so geared motor with motor controller, infrared sensor for limit switches, 3d printed ball chain pulleys, the works.  I can get everything for about $30 a shade vs the ridiculous prices they want for already motorized shades.  If I am reading that one right it is a standard rotating blinds not a roller shade.

**Comment by ryancasler** [comment]
I didn't mean it was a solution, just something else to reference.

**Comment by ogiewon** [comment]
@ericterstegen you can take a look at the following example sketch to see one method of implementing local button control. 

https://github.com/DanielOgorchock/ST_Anything/blob/master/Arduino/Sketches/ST_Anything_Relays_Buttons_ESP8266/ST_Anything_Relays_Buttons_ESP8266.ino

**Comment by vseven** [comment]
I tried for many many hours to get both the neopixel library and the fast LED Library working. Any time there was more than 50 or 60 LEDs it would cause my ESP 32 to reboot. I also sent the code to two different people to try it on an esp8266 and they had the same results.  Maybe you will have better luck but it didn't seem possible within the ST_Anything framework due to timing issues.

**Comment by ericterstegen** [comment]
I have FastLED working with 140 LEDs currently. using ESP32 and FastLED from samguyer: https://github.com/samguyer/FastLED.  

The original FastLED has timing/interrupt issues that plagued me like a fool.  It would core dump all over the place.  I am not sure of all the differences in samguyer's code but it looks like it utilizes RMT better in the ESP32.

I also found my power supply is really fussy and cheap.  It has around 5.2v out but hits around 4v after the LEDs are on full power white.  This caused the ESP to drop out (reboot probably) on the 5v in if I didn't have the USB plugged in.  I added a buck converter to supply the 3.3v on the esp instead of powering the 5v directly.  I am guessing that should isolate the esp from all the crazy voltage swings on my PS.  I am not sure if that is common on the cruddy 5v PS but def was throwing me for a loop.  I am going to update my fork with the new code but wanted to get it stable first.  I am still working out all the other code like what to set when it first powers on and how to handle some of the status stuff.  Still very new to all the ST complexities but it is coming along.
